Reorder summary in order of importance (#12433)
vitessio/vitess: doc/releasenotes/16_0_0_summary.md

### Table of Contents

- **[Major Changes](#major-changes)**
  - **[Breaking Changes](#breaking-changes)**
    - [VTGate Advertised MySQL Version](#advertised-mysql-version)
    - [Default MySQL version on Docker](#default-mysql-version)

>     - [Running Vitess on the Operator](#running-vitess-on-the-operator)

Reviewer: ```suggestion
    - [Running Vitess with the Operator](#running-vitess-on-the-operator)
```

Reviewer: Now that I've read this section, we should make this stronger
Something like `Upgrading to this release with vitess-operator`
or add ⚠️ at the beginning and end!!

>     - [`lock-timeout` and `remote_operation_timeout` Changes](#lock-timeout-introduction)

Reviewer: which binaries do these flags go on?

Reviewer 2: Those are topo flags, they are used by several binaries: `vtgate`, `vttablet`, `vtctld`, `vtbackup`, `vtgr`, `vtorc`, and `vtexplain`.

    - [Orchestrator Integration Deletion](#orc-integration-removal)
    - [vtctld UI Removal](#vtcltd-ui-removal)
    - [vtctld Flag Deprecation & Deletions](#vtctld-flag-deprecations)

>     - [mysqlctl Flags](#mysqlctl-flags)

Reviewer: ```suggestion
    - [mysqlctl Flag Deletions](#mysqlctl-flags)
```

>     - [mysqlctl Flags](#mysqlctl-flags)

Reviewer 2: I think we should have an unified subsection for flag deletion. As of now, we have several subsections:
- `vtctld Flag Deprecation & Deletions`
- `mysqlctl Flags`
- `Removed flag`
- vtorc's `Flag Deprecations`

If we had only 1 sections `Flag Deletions and Deprecations` under `Breaking Changes` and then have sub category per binary, it would be easier for the reader.

    - [Query Serving Errors](#qs-errors)
    - [Logstats Table and Keyspace removed](#logstats-table-keyspace)
    - [Removed Stats](#removed-stats)
    - [Deprecated Stats](#deprecated-stats)

>     - [Removed flag](#removed-flag)

Reviewer: ```suggestion
    - [vttablet Flag Deletions](#removed-flag)
```

## <a id="major-changes"/>Major Changes

### <a id="breaking-changes"/>Breaking Changes

#### <a id="advertised-mysql-version"/>VTGate Advertised MySQL Version

> Since [Pull Request #11989](https://github.com/vitessio/vitess/pull/11989), VTGate advertises MySQL version 8.0.30. This is a breaking change for clients that rely on the VTGate advertised MySQL version and still use MySQL 5.7.

Reviewer 3: it is `8.0.31`

Author: IIRC, we downgraded from 8.0.31 to 8.0.30 because xtrabackup didn't have support for 8.0.31 yet.

Author: This is the PR - https://github.com/vitessio/vitess/pull/12273

Reviewer 2: @GuptaManan100 Correct

Reviewer 3: xtrabackup for the specific purpose has a flag called `–no-server-version-check` this will make xtrabackup to work on 8.0.31 as well. To my knowledge there is not regression of sorts.

5. Remove `innodb_fast_shutdown=0` from your extra cnf in your YAML file.
6. Apply this file.

#### <a id="lock-timeout-introduction"/>`lock-timeout` and `remote_operation_timeout` Changes

Earlier, the shard and keyspace locks used to be capped by the `remote_operation_timeout`. This is no longer the case and instead a new flag called `lock-timeout` is introduced.

> For backward compatibility, if `lock-timeout` is unspecified and `remote_operation_timeout` flag is provided, then its value will also be used for `lock-timeout` as well.

Reviewer: ```suggestion
For backward compatibility, if `lock-timeout` is unspecified and `remote_operation_timeout` flag is provided, then its value will also be used for `lock-timeout`.
```

The default value for `remote_operation_timeout` has also changed from 30 seconds to 15 seconds. The default for the new flag `lock-timeout` is 45 seconds.

During upgrades, if the users want to preserve the same behaviour as previous releases, then they should provide the `remote_operation_timeout` flag explicitly before upgrading.
After the upgrade, they should then alter their configuration to also specify `lock-timeout` explicitly.

#### <a id="orc-integration-removal"/>Orchestrator Integration Deletion

Orchestrator integration in `vttablet` was deprecated in the previous release and is deleted in this release.

> Consider using `VTOrc` instead of `Orchestrator`.

Reviewer: ```suggestion
`VTOrc` should be deployed instead.
```

Reviewer 2: Good point @gedgar, I am adding both links

The following flag is removed in v16:
- `enable_semi_sync`

#### <a id="normalized-lables"/>Normalized labels in the Prometheus Exporter

The Prometheus metrics exporter now properly normalizes _all_ label names into their `snake_case` form, as it is idiomatic for Prometheus metrics. Previously, Vitess instances were emitting inconsistent labels for their metrics, with some of them being `CamelCase` and others being `snake_case`.

### <a id="repl-manager-removal"/>Replication manager removal and VTOrc becomes mandatory

> VTOrc is now a **required** component of Vitess starting from v16. If the users want VTOrc to manage replication, then they must run VTOrc.

Reviewer: ```suggestion
VTOrc is now a **required** component of Vitess starting from v16. If the users want Vitess to manage replication, then they must run VTOrc.
```

Replication manager is removed from vttablets since the responsibility of fixing replication lies entirely with VTOrc now.

> The flag `disable-replication-manager` is deprecated and will be removed in a later release.

Reviewer: ```suggestion
The flag `disable-replication-manager` is deprecated and will be removed in a future release.
```

### <a id="vreplication"/>VReplication

#### <a id="vstream-copy-resume"/>VStream Copy Resume

> In [PR #11103](https://github.com/vitessio/vitess/pull/11103) we introduced the ability to resume a `VTGate` [`VStream` copy operation](https://vitess.io/docs/design-docs/vreplication/vstream/vscopy/). This is useful when a [`VStream` copy operation](https://vitess.io/docs/design-docs/vreplication/vstream/vscopy/) is interrupted due to e.g. a network failure or a server restart. The `VStream` copy operation can be resumed by specifying each table's last seen primary key value in the `VStream` request. Please see the [`VStream` docs](https://vitess.io/docs/16.0/reference/vreplication/vstream/) for more details.

Reviewer: ```suggestion
We now have the ability to resume a `VTGate` [`VStream` copy operation](https://vitess.io/docs/design-docs/vreplication/vstream/vscopy/). This is useful when a [`VStream` copy operation](https://vitess.io/docs/design-docs/vreplication/vstream/vscopy/) is interrupted due to e.g. a network failure or a server restart. The `VStream` copy operation can be resumed by specifying each table's last seen primary key value in the `VStream` request. Please see the [`VStream` docs](https://vitess.io/docs/16.0/reference/vreplication/vstream/) for more details.
```

Reviewer 4: Since we're here, we can update the link: https://vitess.io/docs/16.0/reference/vreplication/vstream/

There's a redirect in place currently.

#### <a id="vdiff2-ga"/>VDiff2 GA

> We are marking [VDiff v2](https://vitess.io/docs/16.0/reference/vreplication/vdiff2/) as Generally Available or production-ready in v16. We now recommend that you use v2 rather than v1 going forward. V1 will be deprecated and eventually removed in future releases.

Reviewer 2: ```suggestion
We are marking [VDiff v2](https://vitess.io/docs/16.0/reference/vreplication/vdiff2/) as production-ready in v16. We now recommend that you use v2 rather than v1 going forward. V1 will be deprecated and eventually removed in future releases.
```

If you wish to use v1 for any reason, you will now need to specify the `--v1` flag.

### <a id="tablet-throttler"/>Tablet throttler

The tablet throttler can now be configured dynamically. Configuration is now found in the topo service, and applies to all tablets in all shards and cells of a given keyspace. For backwards compatibility `v16` still supports `vttablet`-based command line flags for throttler ocnfiguration.

It is possible to enable/disable, to change throttling threshold as well as the throttler query.

See https://github.com/vitessio/vitess/pull/11604

### <a id="inc-backup"/>Incremental backup and point in time recovery

In [PR #11097](https://github.com/vitessio/vitess/pull/11097) we introduced native incremental backup and point in time recovery:

> 

Reviewer: Should we include PR links at all? If we do, it seems like we should do them for all the "major changes" in the release notes, or none of them. As it stands today, it is somewhat inconsistent.

Reviewer 2: I have changed the summary to include links to PRs in most of the major changes.

- It is possible to take an incremental backup, starting with last known (full or incremental) backup, and up to either a specified (GTID) position, or current ("auto") position.
- The backup is done by copying binary logs. The binary logs are rotated as needed.
- It is then possible to restore a backup up to a given point in time (GTID position). This involves finding a restore path consisting of a full backup and zero or more incremental backups, applied up to the given point in time.
- A server restored to a point in time remains in `DRAINED` tablet type, and does not join the replication stream (thus, "frozen" in time).

> - It is possible to take incremental backups from different tablets. It is OK to have overlaps in incremental backup contents. The restore process chooses a valid path, and is valid as long as there are no gaps in the backed up binary log content.

Reviewer: It feels like we can condense this and link to the website docs. Assuming we have them 😉

Reviewer 2: I don't believe we have them, https://github.com/vitessio/vitess/pull/11097 is still marked as `NeedsWebsiteDocsUpdate`, cc @shlomi-noach

Reviewer 5: That is correct :sweat:  I'm having hard time writing those docs, and in particular how to make them co-exist with existing binlog-server -based PITR docs.
